src/main/java/roomescape/ReservationController.java

```diff
@@ -0,0 +1,67 @@
+package roomescape;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+
+
+@Controller
+public class ReservationController {
+    private final List<Reservation> reservations= new ArrayList<>();
+    private final AtomicLong index = new AtomicLong(1);
+
+    public ReservationController() {
+    }
+
+    @GetMapping("/reservation")
+    public String reservation(){
+        return "reservation";
+    }
+
+    @GetMapping("/reservations")
+    @ResponseBody
+    public ResponseEntity<List<Reservation>> reservations(){
+        return new ResponseEntity<>(reservations, HttpStatus.OK);
+    }
+
+    @PostMapping("/reservations")
+    public ResponseEntity<Reservation> create(@RequestBody Reservation reservation){
+        if (reservation.getName() == null
+                || reservation.getName().isEmpty()
+                || reservation.getDate() == null
+                || reservation.getDate().isEmpty()
+                || reservation.getTime() == null
+                || reservation.getTime().isEmpty()
+        ) {
+            throw new NotFoundReservationException("입력 형식이 올바르지 않습니다");
+        } else {
+            Reservation newReservation = new Reservation(index.getAndIncrement(), reservation.getName(), reservation.getDate(), reservation.getTime());
+            reservations.add(newReservation);
+            return ResponseEntity.created(URI.create("/reservations/" + newReservation.getId()))
+                    .body(newReservation);
+        }
+    }
```

Comment:
전체적으로 뎁스가 너무 많은거 같아요 메서드를 빼서 좀더 가독성 및 코드를 이쁘게 짜보시는게 어떠세요?
물론 정답은 없어요~~!!
```suggestion
@PostMapping("/reservations")
public ResponseEntity<Reservation> create(@RequestBody Reservation reservation) {
    validateReservation(reservation);
    Reservation newReservation = new Reservation(
            index.getAndIncrement(),
            reservation.getName(),
            reservation.getDate(),
            reservation.getTime()
    );
    reservations.add(newReservation);
    
    return ResponseEntity.created(URI.create("/reservations/" + newReservation.getId()))
                         .body(newReservation);
}

private void validateReservation(Reservation reservation) {
    if (Stream.of(reservation.getName(), reservation.getDate(), reservation.getTime())
              .anyMatch(field -> field == null || field.isEmpty())) {
        throw new NotFoundReservationException("입력 형식이 올바르지 않습니다");
    }
}
```

Reply:
오호! 뎁스를 나눌 생각을 아예 못했네요! 
혹시 `Array.stream()`를 사용하시지 않고 `stream.of()`를 사용하신 이유가 미리 배열이 만들어져있지않은 친구들이기 때문에 사용하신건가요??!